### PR TITLE
Eminence now has it's special move that lets it pass clockwalls

### LINKED ISSUE #3718: Clockcult: Eminence can't pass through clockwork walls
## Reproduction:

1. Become eminence
2. Attempt to pass clockwork wall
3. Nope negatory not happening

### PR DESCRIPTION
**Pull request by KagiyamaWeb**
## About The Pull Request
Resolves #3718 by adding an INCORPOREAL_MOVE_EMINENCE. 
It shares the same logic as jaunt, but makes sure there are no issues if 30 Runic golems try to pass clock walls. Otherwise it is also blocked by salt and holy water 

## Why It's Good For The Game
It allows Eminence to travel inside the Rebee like it is supposed to

## Changelog
:cl:
add: INCORPOREAL_MOVE_EMINENCE to mob_movement.dm, eminence.dm and __DEFINES/mobs.dm
/:cl:

### CODE REVIEW

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -263,15 +265,29 @@
...
+			L.setDir(direct)
 
+		if(INCORPOREAL_MOVE_EMINENCE) //Incorporeal move for emincence. Blocks move like Jaunt but lets it pass through clockwalls
+			var/turf/open/floor/stepTurf = get_step(L, direct)
+			if(stepTurf)
+				for(var/obj/effect/decal/cleanable/food/salt/S in stepTurf)
+					to_chat(L, "<span class='warning'>[S] bars your passage!</span>")
+					return
+				if(stepTurf.flags_1 & NOJAUNT_1)
+					if(!(stepTurf.contents[1].name == "clockwork wall"))
```
Comment (PowerfulBacon) [inline code/modules/mob/mob_movement.dm]: 😞
Preferably check the area type or z-level.

Comment (PowerfulBacon) [reply code/modules/mob/mob_movement.dm]: In fact it doesnt need any checks since the eminence cant interact with anything.

Comment (KagiyamaWeb) [reply code/modules/mob/mob_movement.dm]: I added a check for being in reebe.

**Review by KagiyamaWeb — COMMENTED** [review]
Check for z_level being reebe

**Comment by francinum** [comment]
This was resolved but why were you checking the name???

**Comment by KagiyamaWeb** [comment]
> 
> 
> This was resolved but why were you checking the name???

Was sleepy, wasn't sure how to best check for clockwall in contents

**Comment by francinum** [comment]
Part of me feels like there's a less shitcode way to do this that doesn't involve adding a new global movement mode...

But hey this works.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**